pandas/core/frame.py

```diff
@@ -4806,13 +4806,14 @@ def _arith_op(left, right):
             return ops.dispatch_to_series(this, other, _arith_op)
         else:
             result = _arith_op(this.values, other.values)
-
-        return self._constructor(result, index=new_index, columns=new_columns,
-                                 copy=False)
+            return self._constructor(result,
+                                     index=new_index, columns=new_columns,
+                                     copy=False)
 
     def _combine_match_index(self, other, func, level=None):
         left, right = self.align(other, join='outer', axis=0, level=level,
                                  copy=False)
+        assert left.index.equals(right.index)
```

Comment:
Why do we need these assert statements?  Are they tested anywhere?

Reply:
These assertions are always true.  I added these so I don't have to keep looking back at core.ops (the only place from which these are called) to double-check what's going on.

Reply:
So this is just a sanity check really?  Perhaps we should comment that then?